core/src/test/java/org/testcontainers/junit/CopyFileToContainerTest.java

```diff
@@ -1,25 +1,62 @@
 package org.testcontainers.junit;
 
-import org.junit.Assert;
 import org.junit.Test;
+import org.testcontainers.containers.BindMode;
 import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.SelinuxContext;
 import org.testcontainers.utility.MountableFile;
+
 import java.io.IOException;
+import java.util.Map;
+
+import static org.rnorth.visibleassertions.VisibleAssertions.assertFalse;
+import static org.rnorth.visibleassertions.VisibleAssertions.assertTrue;
 
 public class CopyFileToContainerTest {
     private static String containerPath = "/tmp/mappable-resource/";
     private static String fileName = "test-resource.txt";
 
     @Test
     public void checkFileCopied() throws IOException, InterruptedException {
-        try(
-            GenericContainer container = new GenericContainer("alpine:latest")
-                .withCommand("sleep","3000")
+        try (
+            GenericContainer container = new GenericContainer()
+                .withCommand("sleep", "3000")
                 .withCopyFileToContainer(MountableFile.forClasspathResource("/mappable-resource/"), containerPath)
         ) {
             container.start();
-            String filesList = container.execInContainer("ls","/tmp/mappable-resource").getStdout();
-            Assert.assertTrue(filesList.contains(fileName));
+            String filesList = container.execInContainer("ls", "/tmp/mappable-resource").getStdout();
+            assertTrue("file list contains the file", filesList.contains(fileName));
+        }
+    }
+
+    @Test
+    public void shouldUseCopyForReadOnlyClasspathResources() throws Exception {
+        try (
+            GenericContainer container = new GenericContainer()
+                .withCommand("sleep", "3000")
+                .withClasspathResourceMapping("/mappable-resource/", containerPath, BindMode.READ_ONLY)
+        ) {
+            container.start();
+            String filesList = container.execInContainer("ls", "/tmp/mappable-resource").getStdout();
+            assertTrue("file list contains the file", filesList.contains(fileName));
         }
     }
+
+    @Test
+    public void shouldUseCopyOnlyWithReadOnlyClasspathResources() {
+        String resource = "/test_copy_to_container.txt";
+        GenericContainer<?> container = new GenericContainer<>()
```

Comment:
Just curious - why the generics?

Reply:
so that `getCopyToFileContainerPathMap()` returns generic Map